src/Widgets/TerminalWidget.vala

```diff
@@ -138,13 +179,20 @@ namespace Terminal {
                     menu.select_first (false);
                     menu.popup_at_pointer (event);
 
-                    return true;
+                    return Gdk.EVENT_STOP;
                 }
 
-                return false;
+                return Gdk.EVENT_PROPAGATE;
+            });
+
+            key_release_event.connect (() => {
```

Comment:
Wouldn't it be better to use the `cursor_moved` signal here? This should cover all cases.
```vala
cursor_moved.connect (remember_position ());
```

Reply:
I don't think so.  We only want to remember the cursor position after the last key press, not when the cursor is moved for other reasons (e.g. due to output from a process).